core/src/main/scala/kafka/log/Log.scala

```diff
@@ -507,6 +507,10 @@ class Log(val dir: File,
     // attempt to read beyond the log end offset is an error
     if(startOffset > next || entry == null)
       throw new OffsetOutOfRangeException("Request for offset %d but we only have log segments in the range %d to %d.".format(startOffset, segments.firstKey, next))
+    
+    maxOffset.foreach((offset) => 
```

Comment:
Sure, I'll move it.

We're running with unclean leader elections disabled, and have seen fetches for 1 past the maxOffset, so I don't know if something related was fixed since 0.8.2.2. There may also just be a bug somewhere in the Storm KafkaSpout.